x/quarantine/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,553 @@
+					newARE(addr0Str, addr3Str, quarantine.AUTO_RESPONSE_UNSPECIFIED),
+				},
+				Pagination: nil,
+			},
+		},
+		{
+			name: "to and same from",
+			req: &quarantine.QueryAutoResponsesRequest{
+				ToAddress:   addr0Str,
+				FromAddress: addr0Str,
+			},
+			resp: &quarantine.QueryAutoResponsesResponse{
+				AutoResponses: []*quarantine.AutoResponseEntry{
+					newARE(addr0Str, addr0Str, quarantine.AUTO_RESPONSE_ACCEPT),
+				},
+				Pagination: nil,
+			},
+		},
+		{
+			name: "only to with no entries",
+			req:  &quarantine.QueryAutoResponsesRequest{ToAddress: addr1Str},
+			resp: &quarantine.QueryAutoResponsesResponse{
+				AutoResponses: nil,
+				Pagination: &query.PageResponse{
+					NextKey: nil,
+					Total:   0,
+				},
+			},
+		},
+		{
+			name: "only to with one entry accept",
+			req:  &quarantine.QueryAutoResponsesRequest{ToAddress: addr2Str},
+			resp: &quarantine.QueryAutoResponsesResponse{
+				AutoResponses: []*quarantine.AutoResponseEntry{
+					newARE(addr2Str, addr3Str, quarantine.AUTO_RESPONSE_ACCEPT),
+				},
+				Pagination: &query.PageResponse{
+					NextKey: nil,
+					Total:   1,
+				},
+			},
+		},
+		{
+			name: "only to with one entry decline",
+			req:  &quarantine.QueryAutoResponsesRequest{ToAddress: addr3Str},
+			resp: &quarantine.QueryAutoResponsesResponse{
+				AutoResponses: []*quarantine.AutoResponseEntry{
+					newARE(addr3Str, addr2Str, quarantine.AUTO_RESPONSE_DECLINE),
+				},
+				Pagination: &query.PageResponse{
+					NextKey: nil,
+					Total:   1,
+				},
+			},
+		},
+		{
+			name: "only to with two entries", // 0 = 1 2 (accept/decline)
+			req:  &quarantine.QueryAutoResponsesRequest{ToAddress: addr0Str},
+			resp: &quarantine.QueryAutoResponsesResponse{
+				AutoResponses: []*quarantine.AutoResponseEntry{
+					newARE(addr0Str, addr1Str, quarantine.AUTO_RESPONSE_ACCEPT),
+					newARE(addr0Str, addr2Str, quarantine.AUTO_RESPONSE_DECLINE),
+				},
+				Pagination: &query.PageResponse{
+					NextKey: nil,
+					Total:   2,
+				},
+			},
+		},
+		{
+			name: "only to with page req",
+			req: &quarantine.QueryAutoResponsesRequest{
+				ToAddress: addr6Str,
+				Pagination: &query.PageRequest{
+					Key:        nil,
+					Offset:     2,
+					Limit:      4,
+					CountTotal: true,
+					Reverse:    false,
+				},
+			},
+			resp: &quarantine.QueryAutoResponsesResponse{
+				AutoResponses: []*quarantine.AutoResponseEntry{
+					newARE(addr6Str, addr2Str, quarantine.AUTO_RESPONSE_DECLINE),
+					newARE(addr6Str, addr3Str, quarantine.AUTO_RESPONSE_ACCEPT),
+					newARE(addr6Str, addr5Str, quarantine.AUTO_RESPONSE_ACCEPT),
+				},
+				Pagination: &query.PageResponse{
+					NextKey: nil,
+					Total:   5,
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			resp, err := s.keeper.AutoResponses(s.stdlibCtx, tc.req)
+			if s.AssertErrorContents(err, tc.err, "AutoResponses error") {
+				s.Assert().Equal(tc.resp, resp, "AutoResponses response")
+			}
+		})
+	}
+}
```

Comment:
The nesting is just struct initialization. Changing to helper funcs for those hurts readability and maintainability because you destroy the field <-> value association. Using struct initialization also makes it so that only the things important to the test are included in the code. With helper functions, you'd end up with a tangled mess of nils and other defaults that just make it harder to figure out what's what, and what's important.